### PR TITLE
Fixed : Disable Post via Email

### PR DESCRIPTION
**Pull request by narenin**
Trac ticket: https://core.trac.wordpress.org/ticket/61144

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/options.php [hunk 1]
```diff
@@ -169,6 +169,17 @@
...
+ * Filters whether the post-by-email functionality is enabled.
+ *
+ * @since 3.0.0
+ *
+ * @param bool $enabled Whether post-by-email configuration is enabled. Default true.
+ */
+if ( apply_filters( 'enable_post_by_email_configuration', true ) ) {
+	$allowed_options['writing'] = array_merge( $allowed_options['writing'], $mail_options );
+}
+
```
Comment (ironprogrammer) [inline src/wp-admin/options.php]: What do you think about replacing [line 161](https://github.com/WordPress/wordpress-develop/blob/3e3b4f61309bbb78ac4e11b2f4ca85dda3efd726/src/wp-admin/options.php#L161) with the filter conditional and assigning the `$allowed_options['writing']` values within like the other conditionals do? I think that might read more clearly and consistently, and not have to merge arrays (maybe ever-so-slightly faster?)

File: src/wp-admin/options.php [hunk 2]
```diff
@@ -179,11 +190,8 @@
 
 	$allowed_options['general'][] = 'users_can_register';
 	$allowed_options['general'][] = 'default_role';
-
-	$allowed_options['writing']   = array_merge( $allowed_options['writing'], $mail_options );
 	$allowed_options['writing'][] = 'ping_sites';
-
-	$allowed_options['media'][] = 'uploads_use_yearmonth_folders';
+	$allowed_options['media'][]   = 'uploads_use_yearmonth_folders';
```
Comment (ironprogrammer) [inline src/wp-admin/options.php]: I don't think these changes are in the scope of this update and would suggest leaving them out for now. The groupings here look intentional.

**Review by ironprogrammer — COMMENTED** [review]
Thanks, @narenin, this is shaping up! A couple of comments here.

**Comment by narendraie** [comment]
Hi @ironprogrammer 

Thanks for the feedback, I have implemented the suggestion can you please check.

File: src/wp-admin/options.php (hunk 1 above)
Comment (narendraie) [reply src/wp-admin/options.php]: done

File: src/wp-admin/options.php (hunk 2 above)
Comment (narendraie) [reply src/wp-admin/options.php]: done

**Comment by narenin** [comment]
Hi @ironprogrammer

Thanks for the feedback, I have implemented the suggestion can you please check.

File: src/wp-admin/options.php (hunk 2 above)
Comment (narenin) [reply src/wp-admin/options.php]: done

File: src/wp-admin/options.php (hunk 1 above)
Comment (narenin) [reply src/wp-admin/options.php]: done

File: src/wp-admin/options.php [hunk 3]
```diff
@@ -179,8 +191,6 @@
 
 	$allowed_options['general'][] = 'users_can_register';
 	$allowed_options['general'][] = 'default_role';
-
```
Comment (ironprogrammer) [inline src/wp-admin/options.php:182]: One nit-pick -- I'd leave this blank line between general and writing.

**Review by ironprogrammer — APPROVED** [review]
This is very succinct, well done! Aside from one small suggested nit, this LGTM.

File: src/wp-admin/options.php (hunk 3 above)
Comment (narenin) [reply src/wp-admin/options.php:182]: @ironprogrammer Thanks for the feedback, I have implemented the suggestions.

**Comment by peterwilsoncc** [comment]
Merged in [r58963](https://core.trac.wordpress.org/changeset/58963) / https://github.com/WordPress/wordpress-develop/commit/0ac256d5fb63f8f1187d37a067168132390a10ab

**Pull request closed without merging**